stories/demos/CommonFeatures.tsx

```diff
@@ -17,6 +17,17 @@ function CurrencyFormatter({ value }: { value: number }) {
   return <>{currencyFormatter.format(value)}</>;
 }
 
+interface SummaryRow {
+  id: string;
+  totalCount: number;
+  availableCount: {
+    yes: number;
+    no: number;
+  };
+}
+
+type SummaryRowFormatterProps = SummaryFormatterProps<SummaryRow, Row>;
```

Comment:
Eventually it ends up like this. However, I still don't think we need to put the `SummaryRow` generic type in the `DataGrid` definition. Unlike regular `Row` which is associated with `Column` definition, the SummaryRow is completely siloed.

Reply:
May be `DataGrid<R, K, SR = unknown>` is not a bad idea. Let's go with the current approach and we can revisit

Reply:
`DataGrid` level really doesn't need to know what does the `summaryRow` looks like as we directly pass the `summaryRow` to the formatter. Maybe we can see if we want to expend the Column type like `ColumnValue<TRow, , TField extends keyof TRow = keyof TRow, TSummaryRow - unknown>`. then we can do this like 

```
const columns: Column: readonly Column<Row, keyof Row, SummaryRow>[] = [
  { 
     ... ,
    summaryFormatter({ row }) {  }
  }
];
```

However, comparing this approach vs not even letting `Column` know the SR types, IMHO, having a separate types like `type SummaryRowFormatterProps = SummaryFormatterProps<SummaryRow, Row>;` is still better.